packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx

```diff
@@ -27,7 +29,7 @@ export interface PanelChromeProps {
   padding?: PanelPadding;
   title?: string;
   titleItems?: PanelChromeInfoState[];
-  menu?: ReactNode;
+  menu?: React.ReactElement;
```

Comment:
We need this to be behind a render prop (render callback), it is expensive to know all the menu actions (might require loading the panel plugin etc, figuring out actions and permissions, might need to check current data state state). 

We can do all that when the menu opens instead if we put this behind a render function. I think that is how the current menu works (only evaluated when you open it). 

The Dropdown component the overlay prop takes a function as well, and it is only called when the overlay is opened

Reply:
Awesome! Thank you 🙏 

Opened an issue for it here: https://github.com/grafana/grafana/issues/59085, so we can tackle it separately (it would be much easier for review).